community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -157,6 +158,26 @@
      */
     Type transactionType();
 
+    /**
+     * Return transaction id if it was already assigned or {@link TransactionData#UNASSIGNED_TRANSACTION_ID} otherwise.
+     * Transaction id is assigned as soon as transaction committed.
+     * @return transaction id.
+     */
+    default long getTransactionId()
```

Comment:
When is the default implementation used? It seems to me that since this interface is an internal one, we should always be able to provide an implementation.